runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/data/GrpcDataService.java

```diff
@@ -53,8 +54,11 @@
     implements FnService, FnDataService {
   private static final Logger LOG = LoggerFactory.getLogger(GrpcDataService.class);
 
-  public static GrpcDataService create(ExecutorService executor) {
-    return new GrpcDataService(executor);
+  public static GrpcDataService create(
+      ExecutorService executor,
+      Function<StreamObserver<BeamFnApi.Elements>, StreamObserver<BeamFnApi.Elements>>
```

Comment:
Use a StreamObserverClientFactory instead of a function.

Reply:
Done (roughly).